src/stan/model/indexing/assign_varmat.hpp

```diff
@@ -205,24 +178,21 @@ inline void assign(Vec1&& x, const Vec2& y, const char* name,
           x.adj().coeffRef(x_idx[i]) = 0.0;
         }
       }
-      using stan::math::forward_as;
-      using stan::math::promote_scalar_t;
-      using stan::math::var;
-      if (!is_constant<Vec2>::value) {
-        for (Eigen::Index i = 0; i < x_idx.size(); ++i) {
-          if (likely(x_idx[i] != -1)) {
-            forward_as<promote_scalar_t<var, Vec2>>(y).adj().coeffRef(i)
-                += prev_vals.coeffRef(i);
-          }
+      for (Eigen::Index i = 0; i < x_idx.size(); ++i) {
+        if (likely(x_idx[i] != -1)) {
+          math::forward_as<math::promote_scalar_t<math::var, Vec2>>(y)
+              .adj()
+              .coeffRef(i)
+              += prev_vals.coeff(i);
```

Comment:
[optional] This could also be simplified with `adjoint_of` if we added `coeffRef` to its `nonexisting_adjoint` return type. I know it is another repo ...

Reply:
I agree that would be v nice, let's not do that in this PR but I'll make a math PR to do that and update it here